 # Genomic data quality control
 ## Description 
 This repository includes the codes used in:\
&nbsp;&nbsp;&nbsp;&nbsp;1. calculating the target coverage (read depth) from the targeted next-generation sequencing (NGS) technique\
&nbsp;&nbsp;&nbsp;&nbsp;2. benchmarking variant calls from a variant calling pipeline using [hap.py](https://www.illumina.com/products/by-type/informatics-products/basespace-sequence-hub/apps/hap-py-benchmarking.html)\
&nbsp;&nbsp;&nbsp;&nbsp;3. plotting the results from 2. using R
 
 
 ## Dependencies
 These codes are based on R 4.0.2 and shell scripts and are executed in the Linux environment.
 
 
 ## Scripts
 #### Extract_TargetCoverage.sh
   - Input: text files generated by [CollectHSMetrics](https://gatk.broadinstitute.org/hc/en-us/articles/360036856051-CollectHsMetrics-Picard-), which record the median, mean and other target coverages of the sequenced samples
   - Function: to ascertain the general sequencing quality of targeted panel NGS 
   - Output: collect the median and mean target coverages of *GJB2*, *SLC26A4*, *MYO15A*, and *OTOF* regions 
 #### ROCcurve_Plotting.R (by Peter Krusche <pkrusche@illumina.com>)
   - Function: run on a set of hap.py results and plot ROCs and PASS points
 #### ROCcurve_Plotting.sh
   - Function: initiate ROCcurve_Plotting.R and store the output files
   - Output: ROC curves of SNVs and Indels in png images
 #### hap.py_ShellScript.sh
   - Input: 
     - a VCF generated by the tested variant calling pipeline
     - a truth VCF (e.g. HG001)
   - Function: conduct [hap.py](https://www.illumina.com/products/by-type/informatics-products/basespace-sequence-hub/apps/hap-py-benchmarking.html) based on the b37 reference genome
   - Output: csv files listing
